rocketpy/motors/TankGeometry.py

```diff
@@ -0,0 +1,376 @@
+    """
+
+    def __init__(self, geometry_dict=dict()):
+        """Initialize TankGeometry class.
+
+        Parameters
+        ----------
+        geometry_dict : dict, optional
+            Dictionary containing the geometry of the tank. The geometry is
+            calculated by a PiecewiseFunction. Hence, the dict keys are disjoint
+            tuples containing the lower and upper bounds of the domain of the
+            corresponding Function, while the values correspond to the radius
+            function from a axis of symmetry.
+        """
+        self.geometry = geometry_dict
+        return None
+
+    @property
+    def geometry(self):
+        """
+        The dictionary containing the geometry of the tank.
+
+        Returns
+        -------
+        dict
+            Dictionary containing the geometry of the tank.
+        """
+        return self._geometry
+
+    @geometry.setter
+    def geometry(self, geometry_dict):
+        """
+        Sets the geometry of the tank.
+
+        Parameters
+        ----------
+        geometry_dict : dict
+            Dictionary containing the geometry of the tank.
+        """
+        self._geometry = dict()
+        for domain, function in geometry_dict.items():
+            self.add_geometry(domain, function)
+
+    @funcify_method("Height (m)", "radius (m)", extrapolation="zero")
+    def radius(self):
+        """
+        The radius of the tank as a function of height.
+
+        Returns
+        -------
+        rocketpy.Function
+            Piecewise defined Function of tank radius.
+        """
+        return self.radius
+
+    @cached_property
+    def average_radius(self):
+        """
+        The average radius of the tank.
+
+        Returns
+        -------
+        float
+            Average radius of the tank.
+        """
+        return self.radius.average(self.bottom, self.top)
+
+    @property
+    def bottom(self):
+        """
+        The bottom of the tank. It is the lowest coordinate that belongs to
+        the domain of the geometry.
+
+        Returns
+        -------
+        float
+            Bottom coordinate of the tank.
+        """
+        return min(self._geometry.keys())[0]
+
+    @property
+    def top(self):
+        """
+        The top of the tank. It is the highest coordinate that belongs to
+        the domain of the geometry.
+
+        Returns
+        -------
+        float
+            Top coordinate of the tank.
+        """
+        return max(self._geometry.keys())[1]
+
+    @cached_property
+    def total_height(self):
+        """
+        The total height of the tank, in meters.
+
+        Returns
+        -------
+        float
+            Total height of the tank.
+        """
+        return self.top - self.bottom
+
+    @funcify_method("Height (m)", "Area (m²)", extrapolation="zero")
+    def area(self):
+        """
+        The area of the tank cross section as a function of height.
```

Comment:
You should define what is the cross section. 
I understand that it is the area parallel to the nozzle exit, but not everyone will understand the same.

Reply:
Solved by expanding docstrings on feb8166690df4330c4c3f8f817f118cd102da25b.